### PR TITLE
localstack: init at 0.9.6

### PR DESCRIPTION
**Pull request by mogorman**
###### Motivation for this change
I wanted to play with localstack. others do to https://github.com/NixOS/nixpkgs/issues/50445

this is still a wip but thought id post here if others wanted to play while i get it working

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by jonringer — COMMENTED** [review]
If some of the packages have very specific version needs, I would first:
 - see if the package is following a version scheme such as semver so that you can widen the allowable range to include what the latest is.
 - if it's mutually exclusive (such as a different major version with breaking changes), then I would bring in the whole nix expression under that directory. 
    - E.g you need pyyaml 3.13, then there would be a `python-modules/awscli/pyyaml_3.nix` that would include the expression to make that package.
  - Or if the package is commonly used at a previous version such as pytest_4, then just add it to the top level with the version (E.g pytest_4)

pinning a lot of specific versions will make it really hard to maintain

**Comment by mogorman** [comment]
ya i saw the right way to do it halfway through. will fix more of that later. im more caught on, how to run npm install as post install step.

File: pkgs/development/python-modules/awscli/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{ lib
+, python
+, groff
+, less
+}:
+
+let
+  py = python.override {
```
Comment (FRidh) [inline pkgs/development/python-modules/awscli/default.nix]: No overrides in `python-packages.nix`.

File: pkgs/development/python-modules/localstack/default.nix [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{ stdenv, python }:
+
+
+let
```
Comment (FRidh) [inline pkgs/development/python-modules/localstack/default.nix]: No overrides in `python-packages.nix`.

**Review by FRidh — CHANGES_REQUESTED** [review]
- All the `buildInputs` need to be checked because most are incorrect.
- Tests!

**Comment by FRidh** [comment]
> ya i saw the right way to do it halfway through. will fix more of that later. im more caught on, how to run npm install as post install step.

Don't. You can probably symlink to a package in `nodePackages`.

**Comment by CMCDragonkai** [comment]
What's the status of this?

**Comment by mogorman** [comment]
dont have time to clean this up

**Comment by CMCDragonkai** [comment]
Someone else can take over this PR?

**Pull request closed without merging**